### PR TITLE
add purge API method

### LINKED ISSUE #788: implement new purge API method
- [ ] Add new purge API method. 
- [ ] This method can run only if service status is DEPLOY_FAILED or DESTROY_FAILED or MANUAL_CLEANUP_REQUIRED
- [ ] Service runs destroy of the method if any resources are available in the deployResources. 
- [ ] Ignore errors from destroy, if any and delete the DB entry at the end.

### PR DESCRIPTION
**Pull request by jinyangyang222**
fixes #788

### CODE REVIEW

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 1]
```diff
@@ -344,6 +344,44 @@ public void asyncDestroyService(Deployment deployment, DeployTask deployTask) {
...
+     * @param deployTask deployTask
+     */
+    @Async("taskExecutor")
+    public void asyncPurgeService(Deployment deployment, DeployTask deployTask) {
+        MDC.put(TASK_ID, deployTask.getId().toString());
+        DeployServiceEntity deployServiceEntity =
+                deployServiceStorage.findDeployServiceById(deployTask.getId());
+        ServiceDeploymentState serviceDeploymentState =
+                deployServiceEntity.getServiceDeploymentState();
+        if (serviceDeploymentState == ServiceDeploymentState.DEPLOY_FAILED
```
[the remark this answers is not in the document]
Comment (jinyangyang222) [reply modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:358]: fixed

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 2]
```diff
@@ -344,6 +344,58 @@ public void asyncDestroyService(Deployment deployment, DeployTask deployTask) {
...
+                || serviceDeploymentState == ServiceDeploymentState.DESTROY_FAILED
+                || serviceDeploymentState == ServiceDeploymentState.MANUAL_CLEANUP_REQUIRED) {
+            asyncPurgeService(deployment, deployTask, deployServiceEntity);
+        } else {
+            throw new InvalidServiceStateException(
+                    String.format("Service with ID %s is not in a state requiring purging.",
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:364]: pls change message to - Service %s is not in the state allowed for purging.

File: modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java [hunk 3]
```diff
@@ -344,6 +344,58 @@ public void asyncDestroyService(Deployment deployment, DeployTask deployTask) {
 
     }
 
+    /**
+     * purge the service based on the serviceDeploymentState.
+     *
+     * @param deployment deployment
+     * @param deployTask deployTask
+     */
+    public void purgeService(Deployment deployment, DeployTask deployTask) {
```
Comment (swaroopar) [inline modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/DeployService.java:353]: can you also please add tests for this? one for success case, one for rejection case.

**Review by swaroopar — CHANGES_REQUESTED** [review]
comments added.

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**